src/hooks/useWebSocket.ts

```diff
@@ -77,8 +77,13 @@ export default function useWebSocket({
   const reconnect = useCallback(() => {
     ws.current = new WebSocket(url);
     ws.current.onopen = () => {
-      state.current = 'connected';
-      setConnected(true);
+      const waitForConnected = setInterval(() => {
+        if (!(ws.current?.readyState === WebSocket.CONNECTING)) {
+          state.current = 'connected';
+          clearInterval(waitForConnected);
+          setConnected(true);
+        }
+      }, 5);
```

Comment:
if this bug is really annoying/blocking then commit this, otherwise let me try to look at it this week (I assume I wrote bad react and there's a solution where we don't have to poll)

Reply:
Yeah, I did brief googling and this seemed like the common solution, but we can hold off on it